arbnode/node.go

```diff
@@ -863,6 +868,12 @@ func createNodeImpl(
 			return nil, err
 		}
 	}
+	var messagePruner *MessagePruner
+	if config.MessagePruner.Enable && !config.Caching.Archive {
+		messagePruner = NewMessagePruner(txStreamer, inboxTracker, stakerObj, func() *MessagePrunerConfig { return &configFetcher.Get().MessagePruner })
+	} else if config.MessagePruner.Enable {
+		return nil, errors.New("message pruning is not allowed for archive nodes")
```

Comment:
- We should have this enabled by default except on archive nodes
- We should only create the message pruner if `stakerObj != nil`

Reply:
done.